Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -832,6 +832,108 @@ def list_device_groups_names():
     )
 
 
+def start_tsf_export():
+    """
+    Start export of tech support file (TSF) from PAN-OS:
+    https://docs.paloaltonetworks.com/pan-os/11-0/pan-os-panorama-api/pan-os-xml-api-request-types/export-files-api/export-technical-support-data
+    """
+    params = {
+        'type': 'export',
+        'category': 'tech-support',
+        'key': API_KEY
+    }
+    result = http_request(
+        URL,
+        'GET',
+        params=params
+    )
+    return result
+
+
+def get_tsf_export_status(job_id: str):
+    """
+    Get status of TSF export.
+    """
+    return_results('Getting TSF export status...')
+    params = {
+        'type': 'export',
+        'category': 'tech-support',
+        'action': 'status',
+        'job-id': job_id,
+        'key': API_KEY
+    }
+    result = http_request(
+        URL,
+        'GET',
+        params=params
+    )
+    return result
+
+
+def download_tsf(job_id: str):
+    """
+    Download an exported TSF.
+    """
+    params = {
+        'type': 'export',
+        'category': 'tech-support',
+        'action': 'get',
+        'job-id': job_id,
+        'key': API_KEY
+    }
+    result = http_request(
+        URL,
+        'GET',
+        params=params,
+        is_file=True
+    )
+    return fileResult("tech_support_file.tar.gz", result.content)
+
+
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get('interval_in_seconds', 30)),
+    timeout=arg_to_number(demisto.args().get('timeout', 1200)),
+    requires_polling_arg=False
+)
+def export_tsf_command(args: dict):
+    """
+    Export a TSF from PAN-OS.
+    """
+    if job_id := args.get('job_id'):
+        export_tsf_status = get_tsf_export_status(job_id).get('response', {}).get('result', {})
+        job_status = export_tsf_status.get('job', {}).get('status')
+        return PollResult(
+            response=download_tsf(job_id),
+            continue_to_poll=job_status != 'FIN',  # continue polling if job isn't done
+        )
+    else:  # either no polling is required or this is the first run
+        result = start_tsf_export()
+        job_id = result.get('response', {}).get('result', {}).get('job', '')
+        if job_id:
+            context_output = {
+                'JobID': job_id,
+                'Status': 'Pending'
+            }
+            continue_to_poll = True
+        else:  # no job ID which is unexpected
+            raise DemistoException("Failed to start tech support file export.")
+
+        return PollResult(
+            response=download_tsf(job_id),
+            continue_to_poll=continue_to_poll,
+            args_for_next_run={
+                'job_id': job_id,
+                'polling': True,
+                'interval_in_seconds': arg_to_number(args.get('interval_in_seconds')),
+                'timeout': arg_to_number(args.get('timeout'))
+            },
+            partial_result=CommandResults(
```

Comment:
The arg `hide_polling_outputs` should also be added to the yaml, there are many example in the repo for reference.

Reply:
@jlevypaloalto `partial_result` actually only returns once. I will send you a screenshot of what the output looks like currently. I did remove a different line that was returning on every polling iteration. 

Re: `hide_polling_outputs`, I did not find any occurrences of this grepping through the repo. I also don't see it documented.
`grep -r "hide_polling_outputs" Packs/`

Can you point me to an example of it?

Reply:
My bad, it's `hide_polling_output` without the `s`.

The purpose of `partial_result ` is to return on every iteration, `poll_message` does so only when `hide_polling_output` isn't in the the yaml.

Maybe `partial_result` returned once because the polling completed after one round?

Reply:
I see, two different `PollResult`s are returned. It’s not exactly conventional but in that case partial_result makes sense.